packages/lesswrong/lib/collections/posts/permissions.ts

```diff
@@ -44,6 +49,8 @@ Posts.checkAccess = async (currentUser: DbUser|null, post: DbPost, context: Reso
     return true
   } else if (userOwns(currentUser, post) || userIsSharedOn(currentUser, post) || await userIsPostGroupOrganizer(currentUser, post)) {
     return true;
+  } else if (!currentUser && constantTimeCompare({ correctValue: canonicalLinkSharingKey, unknownValue: unvalidatedLinkSharingKey })) {
```

Comment:
This (correctly) failed a unit test because if `linkSharingKey` is nullish, then presenting an empty link-sharing key would work, and this meant users could access each other's drafts.